File: frontend/src/types/Movimento.ts
export interface Movimento {
  movimentosId: number
  codigoMovimento: number | null
  clientesNome: string | null
  funcionariosNome: string | null
  valorTotal: number | null
  descontoTotal: number | null
  valorPago: number | null
  dataVenda: string | null
  dataPagamento: string | null
}

export interface ItemMovimento {
  ipmId: number
  produtosDescricao: string | null
  produtosCodigoProduto: string | null
  quantidade: number | null
  precoUnitarioDiaVenda: number | null
  precoTotalDiaVenda: number | null
  precoUnitarioAtual: number | null
  precoTotalAtual: number | null
  dataPagamentoItem: string | null
}

export interface MovimentoDetalhe extends Movimento {
  itens: ItemMovimento[]
}

File: backend/FarmaciaEliseu.Api/Models/Movimento.cs
namespace FarmaciaEliseu.Api.Models;

public class Movimento
{
    public int MovimentosId { get; set; }
    public int? CodigoMovimento { get; set; }
    public int? ClientesId { get; set; }
    public string? ClientesNome { get; set; }
    public int? FuncionariosId { get; set; }
    public string? FuncionariosNome { get; set; }
    public decimal? ValorTotal { get; set; }
    public decimal? DescontoTotal { get; set; }
    public decimal? ValorPago { get; set; }
    public DateTime? DataVenda { get; set; }
    public DateTime? DataPagamento { get; set; }
    public DateTime DataCadastro { get; set; }
    public DateTime DataUltimoRegistro { get; set; }
    public bool Deletado { get; set; }
}
